src/linearAlgebra.jl

```diff
@@ -8,7 +8,28 @@ function norm(pv::PartitionedVector, p::Real = 2; kwargs...)
   return _norm
 end
 
-function dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}; kwargs...) where {T}
+dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}; kwargs...) where {T} = dot(pv1, pv2, Val(pv1.simulate_vector), Val(pv2.simulate_vector); kwargs...)
+
+function dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}, ::Val{true}, ::Val{false}; kwargs...) where {T}
+  N = pv1.epv.N
+  acc = 0
+  for i in 1:N
+    eev1 = pv1[i].vec
+    eev2 = pv2[i].vec
+    acc += dot(eev1, eev2)
+  end
+  return acc
+end
+
+dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}, ::Val{false}, ::Val{true}; kwargs...) where {T} = dot(pv2, pv1)
```

Comment:
Maybe that would be slightly easier to read? But agree that what you have should work.
```suggestion
dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}, ::Val{false}, ::Val{true}; kwargs...) where {T} = dot(pv2, pv1, Val(true), Val(false))
```

Reply:
You're right, it will be easier to read.